components/wmtk_components/mesh_decimation/wmtk/components/mesh_decimation/mesh_decimation.cpp

```diff
@@ -0,0 +1,43 @@
+#include "mesh_decimation.hpp"
+#include "internal/MeshDecimation.hpp"
+#include "internal/MeshDecimationOptions.hpp"
+#include "wmtk/components/base/get_attributes.hpp"
+
+
+namespace wmtk::components {
+
+void mesh_decimation(const base::Paths& paths, const nlohmann::json& j, io::Cache& cache)
+{
+    using namespace internal;
+
+    MeshDecimationOptions options = j.get<MeshDecimationOptions>();
+
+    // input
+    std::shared_ptr<Mesh> mesh_in = cache.read_mesh(options.input);
+    Mesh& mesh = static_cast<Mesh&>(*mesh_in);
+    auto pass_through_attributes = base::get_attributes(cache, mesh, options.pass_through);
+
+    // clear attributes
+    {
+        std::vector<attribute::MeshAttributeHandle> keeps = pass_through_attributes;
+        mesh.clear_attributes(keeps);
```

Comment:
You need to add the `tag` to the attributes you want to keep.

Reply:
the tag I want to keep all inside the `pass_through_attributes`